controllers/machine.go

```diff
@@ -501,7 +507,7 @@ func (mrc *machineReconcileContext) createWorkflow() error {
 		},
 		Spec: tinkv1.WorkflowSpec{
 			TemplateRef: mrc.tinkerbellMachine.Name,
-			HardwareRef: mrc.tinkerbellMachine.Spec.HardwareName,
+			HardwareMap: map[string]string{"device_1": hardware.Spec.Metadata.Instance.ID},
```

Comment:
Is the assumption here that hardware.Spec.Metadata.Instance.ID is going to match the MAC address?

I'm wondering if there is a better way to handle this mapping that provides a clearer mapping between Hardware and Workflow.

Maybe something like this (simplified yaml):
```
hardwareRefs:
- placeholder: "device_1",
- hardwareName: mrc.tinkerbellMachine.Spec.HardwareName,
<some way to specify interface index and whether to use IP or MAC for matching>
```

This would potentially give the same flexibility as the current map without losing an easy way to match the hardware associated with a given workflow (especially for the purposes in displaying in `kubectl get`)

Reply:
> Still reading through the changes, but initial reaction here is: should this be a well defined data type rather than a map[string]string here?

 I think this could use a better data type, but that will necessitate changes to the API now defined in Tinkerbell.  

> Is the assumption here that hardware.Spec.Metadata.Instance.ID is going to match the MAC address?

Yes. We should come up with a better way to encode this, but its probably outside the scope of this PR

Reply:
Cleaning this up in the future sounds good to me